libpimeval/src/pimUtils.h

```diff
@@ -41,6 +42,20 @@ namespace pimUtils
   std::string getDirectoryPath(const std::string& filePath);
   bool getEnvVar(const std::string &varName, std::string &varValue);
 
+  template<typename T> 
+  T operandCast(const void* inputOperand, uint8_t numBytes) {
```

Comment:
How about we derive numBytes from the typename T here? If this works, we don't need to specify sizeof() in every function call

Reply:
Maybe call it typeCast

Reply:
Thanks. Updated the code!

Reply:
I feel something is missing here. If the size of inputOperand is less than the size of T, will some garbage value go into the temp variable? Do you think we need to use num bytes of inputOperand but not num bytes of T?

Reply:
The inputOperand and T are the same size in all the use cases.

Reply:
As a code designing principle, a utility function needs to be self contained. It's better to avoid creating unnecessary assumptions on inputs. Once PR #172 is merged, we need to run that full functional testing.